### PR TITLE
Inspection view

### PR DESCRIPTION
**Pull request by mrica-equinor**

### CODE REVIEW

File: backend/api/Controllers/InspectionController.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error while finding an inspection with task Id {taskId}");
+                return StatusCode(StatusCodes.Status500InternalServerError);
+            }
+
+            if (inspection.Id == null) return NotFound($"Could not find Id for Inspection with task ID {taskId}.");
+
+            var inspectionData = await inspectionService.GetInspectionStorageInfo(inspection.Id);
```
Comment (oysand) [inline backend/api/Controllers/InspectionController.cs:49]: Should we use the isarTaskId here and not the inspection id?

File: frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    align-content: flex-start;
+    gap: 8px;
+    align-self: stretch;
+    flex-wrap: wrap;
+`
+
+interface InspectionDialogViewProps {
+    task: Task
+    setInspectionTask: (inspectionTask: Task | undefined) => void
+    tasks: Task[]
```
Comment (Eddasol) [inline frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx:133]: Would it be more descriptive with task -> selectedTask/displayedTask and tasks -> availableTasks or something? Its not very clear what the difference is

File: frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                </StyledDialog>
+            )}
+        </>
+    )
+}
+
+interface InspectionsViewSectionProps {
+    tasks: Task[]
+    setInspectionTask: (inspectionTask: Task | undefined) => void
+    widthValue?: string | undefined
```
Comment (Eddasol) [inline frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx:236]:
```suggestion
    pictureWidth?: string | undefined
```
More descriptive

File: backend/api/Controllers/InspectionController.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    [ApiController]
+    [Route("inspection")]
+    public class InspectionController(
+            ILogger<InspectionController> logger,
+            IInspectionService inspectionService
+        ) : ControllerBase
+    {
+
+        /// <summary>
+        /// Get the inspection image against an isarTaskId
```
Comment (Eddasol) [inline backend/api/Controllers/InspectionController.cs:19]: Not a fan of using "Against" in this description

```suggestion
        /// Lookup the inspection image for task with specified isarTaskId
```

File: backend/api/Controllers/InspectionController.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<Inspection>> GetInspectionImageById([FromRoute] string installationCode, string taskId)
+        {
+            Inspection? inspection;
+            try
+            {
+                inspection = await inspectionService.ReadByIsarTaskId(taskId, readOnly: true);
```
Comment (Eddasol) [inline backend/api/Controllers/InspectionController.cs:37]: If taskId is isarTaskId, maybe it should be represented in the name to make it more readable

File: backend/api/Controllers/InspectionController.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    {
+
+        /// <summary>
+        /// Get the inspection image against an isarTaskId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpGet]
+        [Authorize(Roles = Role.User)]
+        [Route("{installationCode}/{taskId}/taskId")]
```
Comment (Eddasol) [inline backend/api/Controllers/InspectionController.cs:25]:
```suggestion
        [Route("{installationCode}/{taskId}")]
```
Do we need both?

File: frontend/src/components/Pages/MissionPage/TaskOverview/TaskTable.tsx [hunk 7]
```diff
@@ -25,29 +26,43 @@ const StyledTypography = styled(Typography)`
...
+                        <StyledTableCell>{TranslateText('Description')}</StyledTableCell>
+                        <StyledTableCell>{TranslateText('Inspection Types')}</StyledTableCell>
+                        <StyledTableCell>{TranslateText('Status')}</StyledTableCell>
+                    </Table.Row>
+                </Table.Head>
+                <StyledTableBody>
+                    {tasks && <TaskTableRows tasks={tasks} setInspectionTask={setInspectionTask} />}
+                </StyledTableBody>
+            </StyledTable>
+        </>
```
Comment (Eddasol) [inline frontend/src/components/Pages/MissionPage/TaskOverview/TaskTable.tsx:56]:
```suggestion
            <StyledTable>
                <StyledTableCaptionGray>
                    <StyledTypography variant="h2">{TranslateText('Tasks')}</StyledTypography>
                </StyledTableCaptionGray>
                <Table.Head>
                    <Table.Row>
                        <StyledTableCell>#</StyledTableCell>
                        <StyledTableCell>{TranslateText('Tag-ID')}</StyledTableCell>
                        <StyledTableCell>{TranslateText('Description')}</StyledTableCell>
                        <StyledTableCell>{TranslateText('Inspection Types')}</StyledTableCell>
                        <StyledTableCell>{TranslateText('Status')}</StyledTableCell>
                    </Table.Row>
                </Table.Head>
                <StyledTableBody>
                    {tasks && <TaskTableRows tasks={tasks} setInspectionTask={setInspectionTask} />}
                </StyledTableBody>
            </StyledTable>
```

**Comment by Eddasol** [comment]
"Refactor code" isn't a very good commit message and it seem to contain some fixes of earlier commits. Can it be squashed into the relevant commits?

File: backend/api/Controllers/InspectionController.cs (hunk 1 above)
Comment (mrica-equinor) [reply backend/api/Controllers/InspectionController.cs:49]: Not anymore due to IDA change

File: frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx (hunk 3 above)
Comment (mrica-equinor) [reply frontend/src/components/Pages/MissionPage/InpectionView/InspectionView.tsx:236]: Chandes to boolean called dialogView

**Comment by mrica-equinor** [comment]
This has been split into two other PRs

**Pull request closed without merging**